operator/controllers/kyma_controller_helper_test.go

```diff
@@ -221,24 +220,98 @@ func RemoteKymaExists(remoteClient client.Client, kymaName string) func() error
 	}
 }
 
-func getCatalog(
-	clnt client.Client,
-	kyma *v1alpha1.Kyma,
-) (*v1.ConfigMap, error) {
-	catalog := &v1.ConfigMap{}
-	catalog.SetName(controllers.CatalogName)
-	catalog.SetNamespace(kyma.GetNamespace())
-	err := clnt.Get(ctx, client.ObjectKeyFromObject(catalog), catalog)
-	if err != nil {
-		return nil, err
+func ModuleTemplatesExist(clnt client.Client, kyma *v1alpha1.Kyma) func() error {
+	return func() error {
+		for _, module := range kyma.Spec.Modules {
+			template, err := test.ModuleTemplateFactory(module, unstructured.Unstructured{})
+			if err != nil {
+				return err
+			}
+			if err := clnt.Get(ctx, client.ObjectKeyFromObject(template), template); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+}
+
+func ModuleTemplatesLastSyncGenMatches(clnt client.Client, kyma *v1alpha1.Kyma) func() bool {
+	return func() bool {
+		for _, module := range kyma.Spec.Modules {
+			template, err := test.ModuleTemplateFactory(module, unstructured.Unstructured{})
+			if err != nil {
+				return false
+			}
+			if err := clnt.Get(ctx, client.ObjectKeyFromObject(template), template); err != nil {
+				return false
+			}
+			if template.GetAnnotations() == nil {
+				return false
+			}
+			if strconv.FormatInt(template.GetGeneration(), 10) !=
+				template.GetAnnotations()[v1alpha1.LastSyncGenerationRuntime] {
+				return false
+			}
+		}
+
+		return true
+	}
+}
+
+func ModuleTemplatesLabelsCountMatch(
+	clnt client.Client, kyma *v1alpha1.Kyma, count int,
+) func() bool {
+	return func() bool {
+		for _, module := range kyma.Spec.Modules {
```

Comment:
I think we should refactor this check use not `Eventually` but only `Expect`, the `kyma` here is static, right? Then we can refactor the error check to `Expect(err).ShouldNot(HaveOccurred())` so that for troubleshooting a bit easy for spot the place where error occurred, otherwise we only get this function failed, but don't know exactly which line.

Reply:
no, I'm wrong, the actual resource you want to check here is dynamic, then it's better we return the error instead of true to know exactly what error occurred for troubleshooting purpose.